chain/client/src/chunk_validation.rs

```diff
@@ -261,18 +267,76 @@ fn pre_validate_chunk_state_witness(
         )));
     }
 
+    let block = Chain::get_apply_chunk_block_context(
+        epoch_manager,
+        last_chunk_block.header(),
+        &store.get_previous_header(last_chunk_block.header())?,
+        true,
+    )?;
+    let epoch_id = epoch_manager.get_epoch_id(&block.block_hash)?;
```

Comment:
This `epoch_id` corresponds to the block where `main_transition` happens, but we need to validate transactions for the block for which current chunk is proposed. So you should use `get_epoch_id_from_prev_block` and take previous block from `state_witness.chunk_header`